src/wp-includes/blocks/footnotes.php

```diff
@@ -78,3 +82,137 @@ function register_block_core_footnotes() {
 	);
 }
 add_action( 'init', 'register_block_core_footnotes' );
+
+add_action(
+	'wp_after_insert_post',
+	/**
+	 * Saves the footnotes meta value to the revision.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param int $revision_id The revision ID.
+	 */
+	static function( $revision_id ) {
+		$post_id = wp_is_post_revision( $revision_id );
+
+		if ( $post_id ) {
+			$footnotes = get_post_meta( $post_id, 'footnotes', true );
+
+			if ( $footnotes ) {
+				// Can't use update_post_meta() because it doesn't allow revisions.
+				update_metadata( 'post', $revision_id, 'footnotes', $footnotes );
+			}
+		}
+	}
+);
+
+add_action(
+	'_wp_put_post_revision',
+	/**
+	 * Keeps track of the revision ID for "rest_after_insert_{$post_type}".
+	 *
+	 * @param int $revision_id The revision ID.
+	 */
+	static function( $revision_id ) {
+		global $_gutenberg_revision_id;
+		$_gutenberg_revision_id = $revision_id;
+	}
+);
+
+foreach ( array( 'post', 'page' ) as $post_type ) {
+	add_action(
+		"rest_after_insert_{$post_type}",
+		/**
+		 * This is a specific fix for the REST API. The REST API doesn't update
+		 * the post and post meta in one go (through `meta_input`). While it
+		 * does fix the `wp_after_insert_post` hook to be called correctly after
+		 * updating meta, it does NOT fix hooks such as post_updated and
+		 * save_post, which are normally also fired after post meta is updated
+		 * in `wp_insert_post()`. Unfortunately, `wp_save_post_revision` is
+		 * added to the `post_updated` action, which means the meta is not
+		 * available at the time, so we have to add it afterwards through the
+		 * `"rest_after_insert_{$post_type}"` action.
+		 *
+		 * @since 6.3.0
+		 *
+		 * @param WP_Post $post The post object.
+		 */
+		static function( $post ) {
+			global $_gutenberg_revision_id;
+
+			if ( $_gutenberg_revision_id ) {
+				$revision = get_post( $_gutenberg_revision_id );
```

Comment:
Wondering if we should use [wp_get_post_revision](https://developer.wordpress.org/reference/functions/wp_get_post_revision/) here? It implements `get_post`, but has extra revision-specific checks that `get_post` doesn't.

Well, namely

```php
	if ( 'revision' !== $revision->post_type ) {
		return null;
	}
``` 

😄

Reply:
Answered here: https://github.com/WordPress/gutenberg/pull/52870#discussion_r1272951390